clang/lib/Sema/SemaExpr.cpp

```diff
@@ -8332,7 +8332,7 @@ static bool checkDynamicCountPointerAsParameter(Sema &S, FunctionDecl *FDecl,
     if (!ArgInfo.isCountInParamOrCountPointer() && !ArgInfo.isCountInRet() &&
         !ParmInfo.isCountInParamOrCountPointer() && !ParmInfo.isCountInRet())
       continue;
-    assert(!ActualArgExp->isValueDependent());
```

Comment:
Is is it still necessary to remove this, despite checking `!TheCall->getDependence()` before calling `checkDynamicCountPointerAsParameter`? Or is this something else?

Reply:
`!TheCall->getDependence()` implies no argument `isValueDependent()`.  So I think that assertion becomes unreachable.